app/webpacker/components/Results/Rankings/RankingsTable.jsx

```diff
@@ -0,0 +1,186 @@
+import { continents, countries } from '../../../lib/wca-data.js.erb';
+import { personUrl } from '../../../lib/requests/routes.js.erb';
+
+function CountryCell({ country }) {
+  return (
+    <Table.Cell textAlign="left">
+      {country.iso2 && <CountryFlag iso2={country.iso2} />}
+      {' '}
+      {country.name}
+    </Table.Cell>
+  );
+}
+
+function ResultRow({
+  result, competition, rank, isAverage, show, country,
+}) {
+  const attempts = [result.value1, result.value2, result.value3, result.value4, result.value5]
+    .filter(Boolean);
+
+  const bestResult = _.max(attempts);
+  const worstResult = _.min(attempts);
+  const bestResultIndex = attempts.indexOf(bestResult);
+  const worstResultIndex = attempts.indexOf(worstResult);
+
+  return (
+    <Table.Row>
+      {show === 'by region' ? <CountryCell country={country} />
+        : <Table.Cell textAlign="center">{rank}</Table.Cell> }
+      <Table.Cell>
+        <a href={personUrl(result.personId)}>{result.personName}</a>
+      </Table.Cell>
+      <Table.Cell>
+        {formatAttemptResult(result.value, result.eventId)}
+      </Table.Cell>
+      {show !== 'by region' && <CountryCell country={country} />}
+      <Table.Cell>
+        <CountryFlag iso2={competition.country.iso2} />
+        {' '}
+        <a href={`/competition/${competition.id}`}>{competition.cellName}</a>
+      </Table.Cell>
+      {isAverage && (attempts.map((a, i) => (
+        <Table.Cell>
+          { attempts.length === 5
+              && (i === bestResultIndex || i === worstResultIndex)
+            ? `(${formatAttemptResult(a, result.eventId)})` : formatAttemptResult(a, result.eventId)}
+        </Table.Cell>
+      ))
+      )}
+    </Table.Row>
+  );
+}
+
+export default function RankingsTable({
+  rows, competitionsById, isAverage, show,
+}) {
+  const results = useMemo(() => {
+    const isByRegion = show === 'by region';
+    const [rowsToMap, firstContinentIndex, firstCountryIndex] = isByRegion ? rows : [rows, 0, 0];
+
+    return rowsToMap.reduce((acc, result, index) => {
+      const competition = competitionsById[result.competitionId];
+      const { value } = result;
+
+      const lastItem = acc[acc.length - 1];
+      const previousValue = lastItem?.result.value || 0;
+      const previousRank = lastItem?.rank || 0;
+
+      const rank = value === previousValue ? previousRank : index + 1;
+      const tiedPrevious = rank === previousRank;
+
+      let country = countries.real.find((c) => c.id === result.countryId);
+      if (index < firstContinentIndex) {
+        country = { name: I18n.t('results.table_elements.world') };
+      } else if (index >= firstContinentIndex && index < firstCountryIndex) {
+        country = continents.real.find((c) => c.id === country.continentId);
+      }
+
+      acc.push({
+        result,
+        competition,
+        country,
+        rank,
+        tiedPrevious,
+        key: `${result.id}-${show}-${country.name}`,
+      });
+
+      return acc;
```

Comment:
Would this work with `[...acc, { newObjectStuff: 123 }]`?
Not sure whether that introduces performance issues, but I'm usually not a big fan of stateful calls like `push` or `pop`.

Reply:
it would, don't think it's a performance issue considering it's like at most 200 rows